modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/BulkByScrollUsesAllScrollDocumentsAfterConflictsIntegTests.java

```diff
@@ -201,65 +201,70 @@ public void testDeleteByQuery() throws Exception {
         // Ensure that the write thread blocking task is currently executing
         barrier.await();
 
-        final SearchResponse searchResponse = prepareSearch(sourceIndex).setSize(numDocs) // Get all indexed docs
-            .addSort(SORTING_FIELD, SortOrder.DESC)
-            .execute()
-            .actionGet();
-
-        // Modify a subset of the target documents concurrently
-        final List<SearchHit> originalDocs = Arrays.asList(searchResponse.getHits().getHits());
-        int conflictingOps = randomIntBetween(maxDocs, numDocs);
-        final List<SearchHit> docsModifiedConcurrently = randomSubsetOf(conflictingOps, originalDocs);
-
-        BulkRequest conflictingUpdatesBulkRequest = new BulkRequest();
-        for (SearchHit searchHit : docsModifiedConcurrently) {
-            if (scriptEnabled && searchHit.getSourceAsMap().containsKey(RETURN_NOOP_FIELD)) {
-                conflictingOps--;
+        SearchResponse searchResponse = null;
+        try {
```

Comment:
I think we can use assertResponse here too so we don't need to decref in the test:

```
       int conflictingOps = randomIntBetween(maxDocs, numDocs);
       final int finalConflictingOps = conflictingOps;
        final List<SearchHit> docsModifiedConcurrently = new ArrayList<>();
        assertResponse(prepareSearch(sourceIndex).setSize(numDocs) // Get all indexed docs
            .addSort(SORTING_FIELD, SortOrder.DESC), response -> {
            // Modify a subset of the target documents concurrently
            final List<SearchHit> originalDocs = Arrays.asList(response.getHits().getHits());
            docsModifiedConcurrently.addAll(randomSubsetOf(finalConflictingOps, originalDocs));
        });
```

Reply:
There are some subsequent calls to things initialized in the snippet above, which end up going up until the end of the method (including further assertions). So, instead of having a very large wrapper using `assertResponse` or declaring things outside of the consumer, I opted for a try/finally block.  WDYT? 🤔

Reply:
As I said bellow, we should try to call decRef only from ElasticAssertions for maintainability.

Reply:
Ok, fair enough! Updated.

Reply:
As we also have a nested `assertBusy` that throws a generic `Exception`, is it ok to update the signature of `assertCheckedResponse` to throw an `Exception`, or do we want to keep it restricted to only specific types (currently it's `IOException`, `ExecutionException`, `InterruptedException` ) ?

Reply:
I don't think the assertBusy needs to be nested? you just need to collect the result like in my example?

```
docsModifiedConcurrently.addAll(randomSubsetOf(finalConflictingOps, originalDocs));
```